internal/clients/http/ec2/ec2_client.go

```diff
@@ -442,6 +485,52 @@ func getRoleName(arn string) (string, error) {
 	return roleName[1], nil
 }
 
+func getStatementFromAWSDocument(document string) ([]Statement, error) {
+	unescapedDoc, err := url.QueryUnescape(document)
+	if err != nil {
+		return nil, fmt.Errorf("Could not parse url from the Document: %w", err)
+	}
+
+	var policyVersion Version
+	err = json.Unmarshal([]byte(unescapedDoc), &policyVersion)
+	if err != nil {
+		return nil, fmt.Errorf("Could not unmarshal json: %w", err)
+	}
+	return policyVersion.Statement, nil
+}
+
+func listStatements(versions []*iamTypes.PolicyVersion) ([]Statement, error) {
+	var result []Statement
+	for _, version := range versions {
+		policyVersionStatements, err := getStatementFromAWSDocument(*version.Document)
+		if err != nil {
+			return nil, fmt.Errorf("Could not fetch statement from AWS document: %w", err)
+		}
+
+		result = append(result, policyVersionStatements...)
+	}
+	return result, nil
+}
+
+func listMissingPermissions(statements []Statement, expected Statement) []string {
+	presentPermissions := make(map[string]struct{})
+	var missing []string
+	for _, statement := range statements {
+		if statement.Effect != expected.Effect {
+			continue
+		}
```

Comment:
Can you please add a description or links to the PR explaining the effect, and statements logic?

Reply:
I am not sure I follow, we already have an `ec2` folder and it seems pretty empty so far:

![image](https://user-images.githubusercontent.com/49752/223052314-6e84fc42-9a18-4190-b317-aba544ba3011.png)